tests/integration_tests/reports/alert_tests.py

```diff
@@ -15,10 +15,80 @@
 # specific language governing permissions and limitations
 # under the License.
 # pylint: disable=invalid-name, unused-argument, import-outside-toplevel
+from contextlib import nullcontext
+from typing import List, Optional, Union
 
 import pandas as pd
+import pytest
 from pytest_mock import MockFixture
 
+from superset.reports.commands.exceptions import AlertQueryError
+from superset.reports.models import ReportCreationMethod, ReportScheduleType
+from superset.reports.types import ReportScheduleExecutor
+from superset.utils.database import get_example_database
+from tests.integration_tests.test_app import app
+
+
+@pytest.mark.parametrize(
+    "owner_names,creator_name,config,executor",
+    [
+        (["gamma"], None, [ReportScheduleExecutor.SELENIUM], "admin"),
+        (["gamma"], None, [ReportScheduleExecutor.OWNER], "gamma"),
+        (["alpha", "gamma"], "gamma", [ReportScheduleExecutor.CREATOR_OWNER], "gamma"),
+        (["alpha", "gamma"], "alpha", [ReportScheduleExecutor.CREATOR_OWNER], "alpha"),
+        (
+            ["alpha", "gamma"],
+            "admin",
+            [ReportScheduleExecutor.CREATOR_OWNER],
+            AlertQueryError(),
+        ),
+    ],
+)
+def test_execute_query_as_report_executor(
+    owner_names: List[str],
+    creator_name: Optional[str],
+    config: List[ReportScheduleExecutor],
+    executor: Union[str, Exception],
```

Comment:
nit: it may be easier to read if you rename this to `expected_result` makes more sense when condensing usernames and exceptions

Reply:
Good idea! 👍